components/Layout.vue

```diff
@@ -93,6 +98,36 @@ const section = navigation.find((s) =>
 );
 const { toc } = useContent();
 const showTocDropdown = ref(false);
+
+let visibleHeadingIds = reactive(new Set());
+let observer = null;
+onMounted(() => {
+  const observerOptions = {
+    threshold: 0,
+    rootMargin: '-72px',
```

Comment:
This means that a section that is below the header would not be considered visible.